lib/index.js

```diff
@@ -23,19 +23,33 @@ var Intercom = module.exports = integration('Intercom')
   .endpoint('https://api-segment.intercom.io')
   .ensure('settings.apiKey')
   .ensure('settings.appId')
-  .ensure('message.userId')
   .channels(['server']);
 
 /**
+ * Ensure userId or email.
+ */
+
+Intercom.ensure(function(msg){
+  var email = msg.proxy('traits.email') || msg.proxy('properties.email');
+  var user = msg.userId();
+  if (!(email || user)) {
+    return this.invalid(".userId or .email is required");
```

Comment:
not a big deal, but i think this should be `.reject`, since its message related, not settings related: https://github.com/segmentio/integration#ensurefn